sdk/src/assertions/metadata.rs

```diff
@@ -83,29 +93,53 @@ impl Metadata {
                     return false;
                 }
             }
-            if self.label == labels::METADATA && !ALLOWED_FIELDS.contains(&label.as_str()) {
+            if self.label() == labels::METADATA && !ALLOWED_FIELDS.contains(&label.as_str()) {
                 return false;
             }
         }
         true
     }
+
+    /// Get the label for the metadata
+    pub fn get_label(&self) -> &str {
+        self.label()
+    }
 }
 
 impl AssertionJson for Metadata {}
 
 impl AssertionBase for Metadata {
+    const LABEL: &'static str = labels::METADATA;
+    const VERSION: Option<usize> = Some(ASSERTION_CREATION_VERSION);
+
     fn label(&self) -> &str {
-        &self.label
+        match &self.custom_metadata_label {
+            Some(cm) => cm,
+            None => Self::LABEL,
+        }
     }
 
     fn to_assertion(&self) -> Result<Assertion, Error> {
         Self::to_json_assertion(self)
     }
 
     fn from_assertion(assertion: &Assertion) -> Result<Self, Error> {
-        let mut metadata = Self::from_json_assertion(assertion)?;
-        metadata.label = assertion.label().to_owned();
-        Ok(metadata)
+        match assertion.decode_data() {
```

Comment:
why don't you use Self::from_json_asserrtion() here and then assign metadata based on this: 
metadata.abel = (assertion.label() != labels::METADATA)
    .then(|| assertion.label().to_owned());

Reply:
I can do that.